src/libYARP_OS/src/MacOSAPI.mm

```diff
@@ -0,0 +1,19 @@
+#import "yarp/os/impl/MacOSAPI.h"
+
+#import <Foundation/Foundation.h>
+
+void* enableAppNap()
+{
+    NSProcessInfo *processInfo = [NSProcessInfo processInfo];
+    id activity = [processInfo beginActivityWithOptions:NSActivityLatencyCritical | NSActivityUserInitiated
+                                                 reason:@"YARP requires AppNap off"];
```

Comment:
@francesco-romano I remember when we sketched this piece of code that we misunderstood the how to turn AppNap on off. Here the function name suggest that AppNap will be _enabled_, that is background applications will be put in a "sleep" status, but the `processInfo` reason says _"YARP requires AppNap off"_.
What am I getting wrong here?

Reply:
I will check it again.
I only rebased the old branch and opened the PR, so I didn't check everything.

Thanks to point this out

Reply:
No worries! It may be working just fine, but it looks counterintuitive to me :-)